lib/diaspora/shareable.rb

```diff
@@ -57,7 +65,7 @@ def owned_or_visible_by_user(user)
         with_visibility.where(
           visible_by_user(user).or(arel_table[:public].eq(true)
                                      .or(arel_table[:author_id].eq(user.person_id)))
-        ).select("DISTINCT #{table_name}.*")
+        ).distinct_wrapper
```

Comment:
why? If `.distinct` doesn't work for postgres (I haven't tested that), just leave the `.select("DISTINCT #{table_name}.*")`, because the `.distinct` doesn't make something different for mysql.